components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,235 @@
+                .authenticationContext(context)
+                .authenticatorStatus(status)
+                .authenticatedUser(authenticatedUser)
+                .build();
+    }
+
+    /**
+     * Retrieve the audience.
+     *
+     * @param eventUri     Event URI.
+     * @return Audience string.
+     */
+    public SecurityEventTokenPayload buildSecurityEventToken(EventPayload eventPayload, String eventUri)
+            throws IdentityEventException {
+
+        if (eventPayload == null) {
+            throw new IdentityEventException("Invalid event payload input: Event payload input cannot be null.");
+        }
+
+        if (StringUtils.isEmpty(eventUri)) {
+            throw new IdentityEventException("Invalid event URI input: Event URI input cannot be null or empty.");
+        }
+
+        Map<String, EventPayload> eventMap = new HashMap<>();
+        eventMap.put(eventUri, eventPayload);
+
+        return SecurityEventTokenPayload.builder()
+                .iss(getURL())
+                .iat(System.currentTimeMillis())
+                .jti(UUID.randomUUID().toString())
+                .rci(getCorrelationID())
+                .event(eventMap)
+                .build();
+    }
+
+    /**
+     * Get correlation id from the MDC.
+     * If not then generate a random UUID, add it to MDC and return the UUID.
+     *
+     * @return Correlation id
+     */
+    public String getCorrelationID() {
+
+        String correlationID = MDC.get(CORRELATION_ID_MDC);
+        if (StringUtils.isBlank(correlationID)) {
+            correlationID = UUID.randomUUID().toString();
+            MDC.put(CORRELATION_ID_MDC, correlationID);
+        }
+        return correlationID;
+    }
+
+    private void setLocalUserClaimsToAuthenticatedUser(AuthenticatedUser authenticatedUser,
+                                                       AuthenticationContext context) {
+
+        Map<String, String> claimMappings = (Map<String, String>) context.getParameters()
+                .get(Constants.SP_TO_CARBON_CLAIM_MAPPING);
+
+        if (claimMappings == null) {
+            log.debug("No local claim mappings found for the authenticated user from the context.");
+            return;
+        }
+
+        Map<ClaimMapping, String> userAttributes = authenticatedUser.getUserAttributes();
+
+        if (userAttributes == null) {
+            userAttributes = new HashMap<>();
+        }
+
+        for (Map.Entry<ClaimMapping, String> entry : userAttributes.entrySet()) {
+
+            ClaimMapping claimMapping = entry.getKey();
+            Claim localClaim = claimMapping.getLocalClaim();
+
+            if (claimMappings.containsKey(localClaim.getClaimUri())) {
+                localClaim.setClaimUri(claimMappings.get(localClaim.getClaimUri()));
+            }
+        }
+    }
+
+    /**
+     * Get the tenant qualified URL.
+     *
+     * @return Tenant qualified URL.
+     */
+    public String getURL() {
+
+        return getURL(null);
```

Comment:
Let's not pass null and use the other method.

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/248a1a04b34ee30bc9b512092e84c0397489ea35